src/java/org/apache/cassandra/net/Message.java

```diff
@@ -862,9 +879,9 @@ private Header deserializeHeader(DataInputPlus in, InetAddressAndPort peer, int
             long currentTimeNanos = approxTime.now();
             MonotonicClockTranslation timeSnapshot = approxTime.translate();
             long creationTimeNanos = calculateCreationTimeNanos(in.readInt(), timeSnapshot, currentTimeNanos);
-            long expiresAtNanos = getExpiresAtNanos(creationTimeNanos, currentTimeNanos, TimeUnit.MILLISECONDS.toNanos(in.readUnsignedVInt()));
-            Verb verb = Verb.fromId(in.readUnsignedVInt32());
-            int flags = in.readUnsignedVInt32();
+            long expiresAtNanos = getExpiresAtNanos(creationTimeNanos, TimeUnit.MILLISECONDS.toNanos(in.readUnsignedVInt()));
+            Verb verb = Verb.fromId(Ints.checkedCast(in.readUnsignedVInt()));
+            int flags = Ints.checkedCast(in.readUnsignedVInt());
```

Comment:
nit: Is `in.readUnsignedVInt32()` not equivalent to `Ints.checkedCast(in.readUnsignedVInt())`?